### PR TITLE
Fixing ignore set withCredentials false

### PR DESCRIPTION
**Pull request by tonwe**
Explicit  to set a true or false setting.
if (typeof config.withCredentials !== 'undefined') {
      request.withCredentials = !!config.withCredentials;
 }

### CODE REVIEW

**Comment by kimihiro64** [comment]
@LcTwisk Can someone please merge this? An invalid cookie got accidentally placed in cookie jar and now all my api requests fail...

**Comment by waeltken** [comment]
This is looking good! Looks like i need this too... 👍

**Comment by kimihiro64** [comment]
Bump @waeltken @emilyemorehouse 

This has been sitting here for months, we go live in a month and this is still causing issues if I don't hack this patch in every build...

**Comment by gokhandoslu** [comment]
until this pull request completed, you can use the workaround below

create a file named postinstall.sh and add the codes below
```bash
#!/usr/bin/env bash

find ./node_modules/axios/lib/adapters/xhr.js -type f -exec sed -i '' -e 's|Add withCredentials to request if needed|Add withCredentials to request if needed. Explicit to set a true or false|g' {} \;
find ./node_modules/axios/lib/adapters/xhr.js -type f -exec sed -i '' -e 's|if (config.withCredentials) {|if (typeof config.withCredentials !== "undefined") {|g' {} \;
find ./node_modules/axios/lib/adapters/xhr.js -type f -exec sed -i '' -e 's|request.withCredentials = true;|request.withCredentials = !!config.withCredentials;|g' {} \;
```
add a postinstall script into package.json
`"postinstall": "sh postinstall.sh"`

File: lib/adapters/xhr.js [hunk 1]
```diff
@@ -122,11 +122,10 @@ module.exports = function xhrAdapter(config) {
       });
     }
 
-    // Add withCredentials to request if needed
-    if (config.withCredentials) {
-      request.withCredentials = true;
+    // Add withCredentials to request if needed. Explicit to set a true or false
+    if (typeof config.withCredentials !== 'undefined') {
```
Comment (Alanscut) [inline lib/adapters/xhr.js]: please use the `isUndefined` function of utils.
https://github.com/axios/axios/blob/841466416b6851666955113a60ae46830a27003f/lib/utils.js#L84-L86

**Comment by bxjx** [comment]
Bump on this please! Just hit this issue and am having to use this patch (thanks @gokhandoslu !).

@tonwe @Alanscut, happy to redo pull request if it's stuck waiting on that feedback?

**Comment by bxjx** [comment]
If it helps, I've created a new PR with the feedback https://github.com/axios/axios/pull/2591. Thanks to everyone on this thread for solutions!

**Comment by chinesedfan** [comment]
Closed in favor of #2582. Apologize for the late response.

**Pull request closed without merging**